src/pages/networks/panels/CreateLocalPeeringPanel.tsx

```diff
@@ -0,0 +1,157 @@
+import {
+  ActionButton,
+  Button,
+  ScrollableContainer,
+  SidePanel,
+  useNotify,
+  useToastNotification,
+} from "@canonical/react-components";
+import { useState, type FC } from "react";
+import usePanelParams from "util/usePanelParams";
+import * as Yup from "yup";
+import { useFormik } from "formik";
+import NotificationRow from "components/NotificationRow";
+import ResourceLink from "components/ResourceLink";
+import { queryKeys } from "util/queryKeys";
+import { useQueryClient } from "@tanstack/react-query";
+import { useCurrentProject } from "context/useCurrentProject";
+import type { LxdNetwork } from "types/network";
+import { createNetworkPeer } from "api/network-local-peering";
+import NetworkLocalPeeringForm from "../forms/NetworkLocalPeeringForm";
+import type { LocalPeeringFormValues } from "../forms/NetworkLocalPeeringForm";
+import { testDuplicateLocalPeeringName } from "util/networks";
+
+interface Props {
+  network: LxdNetwork;
+}
+
+const CreateLocalPeeringPanel: FC<Props> = ({ network }) => {
+  const panelParams = usePanelParams();
+  const { project } = useCurrentProject();
+  const notify = useNotify();
+  const toastNotify = useToastNotification();
+  const queryClient = useQueryClient();
+  const controllerState = useState<AbortController | null>(null);
+  const closePanel = () => {
+    panelParams.clear();
+    notify.clear();
+  };
+  const networkURL = `/ui/project/${encodeURIComponent(project?.name ?? "")}/network/${encodeURIComponent(network.name)}`;
+  const projectOtherLabel = "Manually enter project";
+  const networkOtherLabel = "Manually enter network";
+
+  const localPeeringSchema = Yup.object().shape({
+    name: Yup.string()
+      .test(
+        ...testDuplicateLocalPeeringName(
+          panelParams.project,
+          network.name,
+          controllerState,
+        ),
+      )
+      .required("Local peering name is required"),
+    targetProject: Yup.string().required("Target project is required"),
+  });
+
+  const handleSuccess = (peerName: string) => {
+    toastNotify.success(
+      <>
+        Local peering{" "}
+        <ResourceLink
+          type={"peering"}
+          value={peerName}
+          to={`${networkURL}/local-peerings`}
+        />{" "}
+        created for network{" "}
+        <ResourceLink type="network" value={network.name} to={networkURL} />.
+      </>,
+    );
+    closePanel();
+  };
+  const formik = useFormik<LocalPeeringFormValues>({
+    initialValues: {
+      name: "",
+      description: "",
+      targetProject: "",
+      targetNetwork: "",
```

Comment:
The default project should be the current project. That way we immediately show the target network selector, which is a bit more intuitive than the current way of hiding the network input initially.